### ISSUE TITLE
`no-undefined-types` false positive with `@alias`

### ISSUE THREAD

**Issue by ehoogeveen-medweb** [gajus/eslint-plugin-jsdoc#1070]
## Expected behavior
No warning as the alias is defined by the thing it's aliasing.

## Actual behavior
The alias triggers warning `2:1  warning  The type 'Bar' is undefined  jsdoc/no-undefined-types`.

## ESLint Config

```json
{
  "root": true,
  "parserOptions": {
    "ecmaVersion": "latest",
    "sourceType": "module"
  },
  "plugins": ["jsdoc"],
  "rules": {
    "jsdoc/no-undefined-types": "warn"
  }
}

```

## ESLint sample

```js
/**
 * @alias Bar
 */
export class Foo {}
```

## Environment

- Node version: v18.16.0
- ESLint version: v8.40.0
- `eslint-plugin-jsdoc` version: v44.2.0 (no warning with v44.1.0)

## Additional context
We're using aliases to differentiate two implementations of the same class (with different dependencies).

**Comment by ehoogeveen-medweb** [comment]
Thanks!
